task3/index.js

```diff
@@ -0,0 +1,85 @@
+/* 1. Створити ФК Human, яка приймає в аргументи об'єкт і створює властивості name, surname та age. ФК Human також містить методи:
+
+getFullName() - віддає стрінгу повного ім'я, утвореного з name, surname,
+
+setFullName(fullName) - розбиває рядок на name, surname
+	
+Властивості surname, name і age мають бути у всіх ФК. Позбавтеся дублювання за допомогою заміни контексту.
+
+2. Створити ФК* Student, яку я зможу використати так:
+
+let s = new Student(info) // де info.marks = [10,9,8,1,10] та всі інші властивості name, age …
+
+У кожного студента повинні бути такі методи
+
+getAverageMark() - повертає середню оцінку
+getMinMark() - повертає найменшу оцінку
+getMaxMark() - повертає максимальну оцінку
+
+3. Створити ФК Teacher яка приймає в аргументи об'єкт і створює проперті group (масив не менше 5-ти студентів створених за допомогою ФК Student) та містить методи:
+
+- getListOfNamesByAverageMark() - віддає масив імен студентів відсортований за найвищою середньою оцінкою.
+- getStudentByName(name) - отримати один об'єкт студента за ім'ям.
+- removeStudentByName(name) - видалити об'єкт студента, знайденого за ім'ям.
+- updateStudentByName(new Student(...), name) - знайти об'єкт студента по name та замінити на student (новий екземпляр ФК Student)s */
+
+/* Підсказки та правила, яких потрібно дотримуватися при виконанні завдання
+
+- Всі "діти" конструктори повинні успадковувати getFullName та setFullName.
+- Всі "діти" конструктори не повинні дублювати створення полів у конструкторі.Позбавтеся дублювання за допомогою заміни контексту.
+- Всі конструктори приймають об'єкт у аргументи, а не аргументи через кому
+- Методи описуємо у прототипі
+- Слідкуйте, щоб ваш код не повторювався - якщо потрібно створюйте додаткові методи і не забувайте користуватися готовими.
+- Перевіряйте ваш код декілька разів
+- Читайте уважно умову завдання.
+*/
+
+function Human({ name, surname, age }) {
+	this.name = name;
+	this.surname = surname;
+	this.age = age;
+
+	this.getFullName = function () {
+		return `${this.name} ${this.surname}`;
+	};
+	this.setFullName = function (fullName) {
+		return ([this.name, this.surname] = fullName.split(' '));
+	};
+}
+
+const user1 = new Human({ name: 'Vladyslav', surname: 'Tsarik', age: 26 });
+
+console.dir(user1);
```

Comment:
your variant
<img width="403" alt="Screenshot 2023-03-30 at 00 32 44" src="https://user-images.githubusercontent.com/38155972/228672771-d872e621-73ae-480c-b452-2ae2a73b5d79.png">
but should be
<img width="372" alt="image" src="https://user-images.githubusercontent.com/38155972/228673533-b4b54a26-3ff2-472d-bf20-2c3f363e74a7.png">